components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -10517,49 +10517,104 @@ private void addAPIPolicy(APIPolicy policy, Connection conn) throws SQLException
-            if (driverName.contains("MS SQL") || driverName.contains("Microsoft")) {
-                st.executeUpdate("SET IDENTITY_INSERT AM_API_THROTTLE_POLICY ON");
+    public APIPolicy updateAPIPolicy(APIPolicy policy) throws APIManagementException {
+        String updateQuery;
+        int policyId = 0;
+        String selectQuery;
+        if (policy != null) {
+            if (!StringUtils.isBlank(policy.getPolicyName()) && policy.getTenantId() != -1) {
+                selectQuery = SQLConstants.ThrottleSQLConstants.GET_API_POLICY_ID_SQL;
+                updateQuery = SQLConstants.ThrottleSQLConstants.UPDATE_API_POLICY_SQL;
+            } else if (!StringUtils.isBlank(policy.getUUID())) {
+                selectQuery = SQLConstants.ThrottleSQLConstants.GET_API_POLICY_ID_BY_UUID_SQL;
+                updateQuery = ThrottleSQLConstants.UPDATE_API_POLICY_BY_UUID_SQL;
+            } else {
+                String errorMsg = "Policy object doesn't contain mandatory parameters. At least UUID or Name,Tenant Id"
+                                + " should be provided. Name: " + policy.getPolicyName()
+                                + ", Tenant Id: " + policy.getTenantId() + ", UUID: " + policy.getUUID();
+                log.error(errorMsg);
+                throw new APIManagementException(errorMsg);
             }
-            String dbProductName = conn.getMetaData().getDatabaseProductName();
-            policyStatement = conn.prepareStatement(addQuery,
-                    new String[]{DBUtils.getConvertedAutoGeneratedColumnName(dbProductName, "POLICY_ID")});
-            setCommonParametersForPolicy(policyStatement, policy);
-            policyStatement.setString(12, policy.getUserLevel());
-            policyStatement.setBoolean(10, true);
-            policyStatement.setInt(13, policyId);
-            int updatedRawCount = policyStatement.executeUpdate();
+        } else {
+            String errorMsg = "Provided Policy to update is null";
+            log.error(errorMsg);
+            throw new APIManagementException(errorMsg);
+        }
 
-            if (driverName.contains("MS SQL") || driverName.contains("Microsoft")) {
-                st.executeUpdate("SET IDENTITY_INSERT AM_API_THROTTLE_POLICY OFF");
-            }
+        try (Connection connection = APIMgtDBUtil.getConnection()) {
+            connection.setAutoCommit(false);
+            try ( PreparedStatement selectStatement = connection.prepareStatement(selectQuery);
+                  PreparedStatement deleteStatement = connection.prepareStatement(SQLConstants
+                          .ThrottleSQLConstants.DELETE_CONDITION_GROUP_SQL);
+                  PreparedStatement updateStatement = connection.prepareStatement(updateQuery)) {
+                if (selectQuery.equals(SQLConstants.ThrottleSQLConstants.GET_API_POLICY_ID_SQL)) {
+                    selectStatement .setString(1, policy.getPolicyName());
```

Comment:
Shall we fix the spaces here?
`selectStatement .setString(`

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/9541/commits/2809e66ce7558218fa0dd2ac5e74a4e82ff5fb83